src/routes/sessions.js

```diff
@@ -45,67 +45,118 @@ module.exports = function (app, opts, dependencies) {
     });
   }
 
-  function sendToken(response, renderingId) {
+  function sendJwtToken(response, renderingId) {
     return function (user) {
-      var token = createToken(user, renderingId);
+      const token = createToken(user, renderingId);
       response.send({ token: token });
     };
   }
 
   function formatAndSendError(response) {
     return function (error) {
-      var body;
+      let body;
       if (error && error.message) {
         body = { errors: [{ detail: error.message }] };
       }
       return response.status(401).send(body);
     };
   }
 
+  function isTwoFactorTokenValid(user, twoFactorToken) {
+    const partialUserSecret = user.partial_user_secret;
+    const userSecret = new UserSecretCreator(partialUserSecret, process.env.LIANA_2FA_SECRET)
+      .perform();
+
+    return otplib.authenticator.verify({ token: twoFactorToken, secret: userSecret });
+  }
+
+  function getTwoFactorRegistrationResponse(user) {
+    if (process.env.LIANA_2FA_SECRET  === undefined) {
```

Comment:
Yes, but after we talk with Sandro (Sandro, you and me), and I think we decided to finally use `LIANA_2FA_SECRET`. But anyway, because we need to have a key contrained to 20 characters to avoid key colisions in the process of combining the partial user secret and the LIANA_2FA_SECRET.